src/aux-records/RecordsServer.ts

```diff
@@ -2090,6 +2140,178 @@ export class RecordsServer {
                     return result;
                 }),
 
+            getXpUserMeta: procedure()
+                .origins('api')
+                .http('GET', '/api/v2/xp/user')
+                .inputs(GetXpUserById)
+                .handler(async (input, context) => {
+                    const authUser = await this._validateSessionKey(
+                        context.sessionKey
+                    );
+                    if (!authUser.success) {
+                        return authUser;
+                    }
+                    if (!input.userId && !input.xpId && !authUser.userId) {
+                        return {
+                            success: false,
+                            errorCode: 'unacceptable_request',
+                            errorMessage:
+                                'One of properties userId or xpId must be provided.',
+                        };
+                    }
+                    if (input.userId && input.xpId) {
+                        return {
+                            success: false,
+                            errorCode: 'unacceptable_request',
+                            errorMessage:
+                                'You cannot provide both userId and xpId.\nProperties are mutually exclusive.',
+                        };
+                    }
+                    //* An empty string for any of the query types will be treated as the current logged in user
+                    const user = await this._xpController.getXpUser(
+                        input.xpId
+                            ? { xpId: input.xpId }
+                            : input.userId
+                            ? { userId: input.userId }
+                            : { userId: authUser.userId }
+                    );
+                    return user;
+                }),
```

Comment:
@LandonSiler If xpController is null or undefined, then this procedure should check that and return a `not_supported` result just like the `getNotificationsApplicationServerKey` procedure.